$OpenBSD: patch-library_rsa_c,v 1.1.1.1 2014/04/18 11:37:02 sthen Exp $

Avoid triggering APIWARN (though using rand() for this self-test is
actually harmless).

--- library/rsa.c.orig	Fri Apr 18 12:01:57 2014
+++ library/rsa.c	Fri Apr 18 12:02:07 2014
@@ -1469,13 +1469,10 @@ void rsa_free( rsa_context *ctx )
 #if defined(POLARSSL_PKCS1_V15)
 static int myrand( void *rng_state, unsigned char *output, size_t len )
 {
-    size_t i;
-
     if( rng_state != NULL )
         rng_state  = NULL;
 
-    for( i = 0; i < len; ++i )
-        output[i] = rand();
+    arc4random_buf(output, len);
 
     return( 0 );
 }
